sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -4713,6 +4713,12 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
     val df6 = df3.join(df2, col("df3.zaak_id") === col("df2.customer_id"), "outer")
     df5.crossJoin(df6)
   }
+
+  test("SPARK-46448: InlineTable column is nullable if converting to Decimal could be null") {
+    val plan = sql("SELECT * FROM values(-0.172787979),(533704665545018957788294905796.5)")
```

Comment:
wait, so we still get null value at the end?

Reply:
This is only a defensive change because `Cast.canNullSafeCastToDecimal(t, d)` still has code to return `false`. though we know that no null value will happen at the end.